Kernel/Modules/AgentTicketCompose.pm

```diff
@@ -1813,59 +2180,244 @@ sub Run {
             }
         }
 
-        # remember dynamic field validation results if erroneous
-        my %DynamicFieldPossibleValues;
+        # get list type
+        my $TreeView = 0;
+        if ( $ConfigObject->Get('Ticket::Frontend::ListType') eq 'tree' ) {
+            $TreeView = 1;
+        }
 
-        # cycle trough the activated Dynamic Fields for this screen
-        DYNAMICFIELD:
-        for my $DynamicFieldConfig ( @{ $Self->{DynamicField} } ) {
-            next DYNAMICFIELD if !IsHashRefWithData($DynamicFieldConfig);
+        my $Autoselect = $ConfigObject->Get('TicketACL::Autoselect') || undef;
 
-            my $PossibleValuesFilter;
+        # gather fields which are supposed to be hidden when autoselected
+        my $HideAutoselectedJSON;
+        if ($Autoselect) {
+            my @HideAutoselected = grep { !ref( $Autoselect->{$_} ) && $Autoselect->{$_} == 2 } keys %{$Autoselect};
+            AUTOSELECT:
+            for my $Item (@HideAutoselected) {
+                next AUTOSELECT if $Item ne 'NextStateID';
+                $Item = 'StateID';
```

Comment:
Please also align the below (not making comments there, see AJAXUpdate), and changing this here should not be necessary then.